File: app/src/main/java/com/johnpetitto/orachat/ui/SimpleTextWatcher.java
package com.johnpetitto.orachat.ui;

import android.text.Editable;
import android.text.TextWatcher;

public abstract class SimpleTextWatcher implements TextWatcher {
    public abstract void onTextChanged(String text);

    @Override
    public final void beforeTextChanged(CharSequence charSequence, int i, int i1, int i2) {}

    @Override
    public final void onTextChanged(CharSequence charSequence, int i, int i1, int i2) {
        onTextChanged(charSequence.toString());
    }

    @Override
    public final void afterTextChanged(Editable editable) {}
}

File: app/src/main/java/com/johnpetitto/orachat/ui/SimpleTextWatcher.kt
package com.johnpetitto.orachat.ui

import android.text.Editable
import android.text.TextWatcher

abstract class SimpleTextWatcher: TextWatcher {
    abstract fun onTextChanged(text: String)

    override final fun afterTextChanged(p0: Editable?) {}
    override final fun beforeTextChanged(p0: CharSequence?, p1: Int, p2: Int, p3: Int) {}
    override final fun onTextChanged(p0: CharSequence?, p1: Int, p2: Int, p3: Int) {
        onTextChanged(p0.toString())
    }
}
